crates/containerd-shim-wasmedge/src/executor.rs

```diff
@@ -98,6 +100,27 @@ impl WasmEdgeExecutor {
 
         Ok(vm)
     }
+
+    fn can_handle(&self, spec: &Spec) -> Result<(), ExecutorValidationError> {
+        // check if the entrypoint of the spec is a wasm binary.
+        let (module_name, _method) = oci::get_module(spec);
+        let module_name = match module_name {
+            Some(m) => m,
+            None => {
+                log::info!("WasmEdge cannot handle this workload, no arguments provided");
+                return Err(ExecutorValidationError::CantHandle(EXECUTOR_NAME));
+            }
+        };
+        let path = PathBuf::from(module_name);
+
+        path.extension()
+            .map(|ext| ext.to_ascii_lowercase())
+            .is_some_and(|ext| ext == "wasm" || ext == "wat")
+            .then_some(())
+            .ok_or(ExecutorValidationError::InvalidArg)?;
```

Comment:
IIUC, this will make it so that if the entrypoint doesn't end in `wasm` or `wat` then the container fails, rather than attempt to run a native container.
I think the logic was inverted before. We first tried to execute as a native container, and if that failed, we tried with as a wasm container.